vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -1298,19 +1383,22 @@ def fused_experts_impl(hidden_states: torch.Tensor,
         else:
             raise ValueError(f"Unsupported FusedMoe activation: {activation}")
 
-        a2q_scale: Optional[torch.Tensor] = None
-
-        if use_fp8_w8a8:
-            qintermediate_cache2, a2q_scale = _fp8_quantize(
-                intermediate_cache2, a2_scale, block_shape)
-        else:
-            qintermediate_cache2 = intermediate_cache2
-            a2q_scale = a2_scale
+        intermediate_cache2, a2_scale = moe_kernel_prepare_input(
```

Comment:
I think you might need to use different variables here since the cache might get reused (if the loop has multiple iterations).

Reply:
Okay sure, I didn't understand why this was required but I believe you

Reply:
You should be able to trigger/test this by setting `envs.VLLM_FUSED_MOE_CHUNK_SIZE` to something smaller to force the chunking loop in fused_moe.py to run more than once.  There's also a problem size in test_moe.py that triggers this too.